src/runtimes/docker/contracts/CommandRunner.ts

```diff
@@ -5,37 +5,57 @@
 
 import { CommandLineArgs } from '../utils/commandLineBuilder';
 
+type CommandResponseBase = {
+    command: string;
+    args: CommandLineArgs;
+};
+
+export type PromiseCommandResponse<T> = CommandResponseBase & {

```

Comment:
The old `CommandResponse` was already returning a promise from `parse()`, so I suggest keeping the old name. That will reduce the number of changes in this PR and also avoid having "promise of a promise" in the signatures, which does not read very comfortably.

Reply:
IMO we should keep it as is; calling it "Promise" is to contrast it from `GeneratorCommandResponse<T>` which has an `AsyncGenerator<T>` return from its parsing method.

Reply:
I understand, but "linguistics" PoV it makes sense to call the most common case just "Foo", and the special case "SpecialFoo". E.g glider vs motor glider. 

I do not feel strongly about this, but still think it would be a simplification.